src/code/z_parameter.c

```diff
@@ -52,38 +332,750 @@
+                if (GET_CUR_EQUIP_VALUE(EQUIP_SWORD) == 0) {
+                    BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) = ITEM_NONE;
+                } else {
+                    BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) =
+                        GET_CUR_EQUIP_VALUE(EQUIP_SWORD) + ITEM_SWORD_KOKIRI - 1;
+                }
+            }
+        }
+    }
+
+    for (btn = 0; btn < 1; btn++) {
+        if ((GET_CUR_FORM_BTN_ITEM(btn) != ITEM_NONE) && (GET_CUR_FORM_BTN_ITEM(btn) != ITEM_UNK_FD)) {
+            Interface_LoadItemIconImpl(globalCtx, btn);
+        }
+    }
+}
+
+s32 Inventory_HasEmptyBottle(void) {
+    s32 slot;
+
+    for (slot = SLOT_BOTTLE_1; slot <= SLOT_BOTTLE_6; slot++) {
+        if (gSaveContext.save.inventory.items[slot] == ITEM_BOTTLE) {
+            return true;
+        }
+    }
+    return false;
+}
+
+s32 Inventory_HasItemInBottle(u8 item) {
+    s32 slot;
+
+    for (slot = SLOT_BOTTLE_1; slot <= SLOT_BOTTLE_6; slot++) {
+        if (gSaveContext.save.inventory.items[slot] == item) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void Inventory_UpdateBottleItem(GlobalContext* globalCtx, u8 item, u8 btn) {
+    gSaveContext.save.inventory.items[GET_CUR_FORM_BTN_SLOT(btn)] = item;
+    SET_CUR_FORM_BTN_ITEM(btn, item);
+
+    Interface_LoadItemIconImpl(globalCtx, btn);
+
+    globalCtx->pauseCtx.cursorItem[0] = item;
+    gSaveContext.buttonStatus[btn] = BTN_ENABLED;
+
+    if (item == ITEM_HOT_SPRING_WATER) {
+        func_8010EBA0(60, GET_CUR_FORM_BTN_SLOT(btn) - SLOT_BOTTLE_1);
+    }
+}
+
+s32 Inventory_ConsumeFairy(GlobalContext* globalCtx) {
+    u8 bottleSlot = SLOT(ITEM_FAIRY);
+    u8 btn;
+    u8 i;
+
+    for (i = 0; i < 6; i++) {
+        if (gSaveContext.save.inventory.items[bottleSlot + i] == ITEM_FAIRY) {
+            for (btn = 1; btn < 4; btn++) {
+                if (GET_CUR_FORM_BTN_ITEM(btn) == ITEM_FAIRY) {
+                    SET_CUR_FORM_BTN_ITEM(btn, ITEM_BOTTLE);
+                    Interface_LoadItemIconImpl(globalCtx, btn);
+                    bottleSlot = GET_CUR_FORM_BTN_SLOT(btn);
+                    i = 0;
+                    break;
+                }
+            }
+            gSaveContext.save.inventory.items[bottleSlot + i] = ITEM_BOTTLE;
+            return true;
+        }
+    }
+
+    return false;
+}
+
+/**
+ * Only used to equip spring water when hot sprint water timer runs out
+ */
+void Inventory_UpdateItem(GlobalContext* globalCtx, s16 slot, s16 item) {
+    s16 btn;
+
+    gSaveContext.save.inventory.items[slot] = item;
+
+    for (btn = 1; btn < 4; btn++) {
+        if (GET_CUR_FORM_BTN_SLOT(btn) == slot) {
+            SET_CUR_FORM_BTN_ITEM(btn, item);
+            Interface_LoadItemIconImpl(globalCtx, btn);
+        }
+    }
+}
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_801153C8.s")
 
+TexturePtr sDoActionTextures[] = {
+    0x09000000, // gAttackDoActionENGTex
+    0x09000180, // gCheckDoActionENGTex
```

Comment:
Could you just do this asset file now? I know it's fairly small. You can always add symbols if you want them for other functions.

Reply:
I might look into it

Reply:
Updated to include these now